public/js/components/AtomEdit/CustomEditors/StoryQuestionFields/Question.js

```diff
@@ -19,29 +23,93 @@ export class StoryQuestionsQuestion extends React.Component {
     onFormErrorsUpdate: PropTypes.func
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      answers: props.fieldValue && props.fieldValue.answers && props.fieldValue.answers.length ?
+        props.fieldValue.answers :
+        []
+    };
+  }
+
+  componentDidMount() {
+    this.state.answers.forEach((answer, i) => {
+    // eslint-disable-next-line
+    const [_, atomType, atomId] = answer.answerId.split('/');
+      fetchCapiAtom(atomType, atomId)
+        .then(atom => {
+          this.setState(prevState => {
+            const answers = prevState.answers.slice();
+            answers[i].title = atom.title;
+          });
+        });
+    });
+  }
+
   updateQuestion = (questionObject) => {
     const questionWithId = Object.assign({}, questionObject, {
-      questionId: questionObject.questionId || uuidv4()
+      questionId: questionObject.questionId || uuidv4(),
+      answers: this.state.answers
     });
 
     this.props.onUpdateField(questionWithId);
   }
 
-  render () {
-    const answersCount = (this.props.fieldValue && this.props.fieldValue.answers) ? this.props.fieldValue.answers.length : 0;
+  onSelect = (snippet) => {
+    this.setState(
+      (oldState) => Object.assign({}, oldState, { answers: oldState.answers.concat({
```

Comment:
We don't need to update the whole state here, sending a new answers block is enough:
``` js
(oldState) => {
  answers: oldState.answers.concat(...etc);
}
```